searchlib/src/vespa/searchlib/aggregation/quantile_aggregation_result.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+#include "quantile_aggregation_result.h"
+
+#include <vespa/searchlib/expression/resultvector.h>
+#include <vespa/vespalib/objects/serializer.hpp>
+#include <vespa/vespalib/objects/deserializer.hpp>
+
+namespace search::aggregation {
+
+using expression::ResultNodeVector;
+using vespalib::Serializer;
+using vespalib::Deserializer;
+
+IMPLEMENT_IDENTIFIABLE_NS2(search, aggregation, QuantileAggregationResult, AggregationResult);
+
+QuantileAggregationResult::QuantileAggregationResult() {
+    _rank.reset(new FloatResultNode(0));
+}
+
+QuantileAggregationResult::QuantileAggregationResult(const ResultNode::CP& result) {
+    setResult(result);
+}
+
+QuantileAggregationResult::~QuantileAggregationResult() = default;
+
+QuantileAggregationResult::QuantileAggregationResult(const QuantileAggregationResult&) = default;
+
+QuantileAggregationResult& QuantileAggregationResult::operator=(const QuantileAggregationResult&) = default;
+
+void QuantileAggregationResult::visitMembers(vespalib::ObjectVisitor& visitor) const {
+    AggregationResult::visitMembers(visitor);
+    visit(visitor, "quantiles", _quantiles);
+
+    for (const auto [quantile, value] : quantile_results()) {
+        visit(visitor, std::format("quantile({})", quantile), value);
+    }
+
+    visit(visitor, "extension", _extension);
+}
+
+void QuantileAggregationResult::onPrepare(const ResultNode& result, bool useForInit) {
+    (void) result;
```

Comment:
consider logging a warning if useForInit is true here (I'm pretty sure that shouldn't happen)